### PR TITLE
set up the template

### PR DESCRIPTION
**Pull request by kalenkovich**
I had to start from scracth. There is still no actual frequency-tagging code added yet. Just making sure everything works, tests are passed, docs are built, etc.

I did pretty much just what is described in the docs for the mne-tools project template [here](https://mne.tools/mne-project-template/quick_start.html). Not everything worked straight away though - thus a thousand commits. Also, I am not sure what half the CI-like services are doing but I connected them anyway. Appveyor failing is fine, by the way: I haven't told it what to actually do yet.

@dominikwelke, could you please check that you can build the package, the docs and the tests are passed locally? I put the instructions under contributing.md.

*Just complaining:* The darndest thing is that it took me an extra million years just because `open` and `codecs.open` behave a bit differently on Windows. Because of that I always got an error when trying to deploy to PyPI. And the error message was so "informative" and the error itself was so common that I found this image long before I actually figured out how to fix it:

### CODE REVIEW

**Comment by kalenkovich** [comment]
There are some pep8 issues - I'll fix those before merging.

**Comment by dominikwelke** [comment]
- [x] setup works
- [x] coverage test works
- [x] tests passed
- [x] doc builds

two things though:
- i'll make some inline suggestions to the instructions in `contributing.md`. 
- creating the conda environment from `environment.yml`: 
i'm not super familiar with it, but if i read the file correctly `numpy`, `sciypy`, and `mne` are fetched twice - once via `conda` in the `environment.yml` file, and once via `pip` in the `requirements.txt` file.  i assume conda will have priority. we could eliminate them from either of the requirement files

File: contributing.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+From root
+
+```
+pip install -e .
+```
+
+## Check test coverage
+
+From `freqtag`:
```
Comment (dominikwelke) [inline contributing.md]: this is a bit confusing since all three the package, the conda environment, and the folder we cd to are called freqtag :)

**Comment by dominikwelke** [comment]
> _Just complaining:_ The darndest thing is that it took me an extra million years just because `open` and `codecs.open` behave a bit differently on Windows. Because of that I always got an error when trying to deploy to PyPI. And the error message was so "informative" and the error itself was so common that I found this image long before I actually figured out how to fix it:
> 
> 

:D

**Comment by kalenkovich** [comment]
> * [x]  setup works
> * [x]  coverage test works
> * [x]  tests passed
> * [x]  doc builds

Thanks for checking!
 
> two things though:
> 
> * i'll make some inline suggestions to the instructions in `contributing.md`.

A bit confused about that one. You've already committed your suggestions, right?

> * creating the conda environment from `environment.yml`:
>   i'm not super familiar with it, but if i read the file correctly `numpy`, `sciypy`, and `mne` are fetched twice - once via `conda` in the `environment.yml` file, and once via `pip` in the `requirements.txt` file.  i assume conda will have priority. we could eliminate them from either of the requirement files

`conda` won't automatically install stuff from `requirements.txt` so there won't be any double installation. I am not entirely sure but I think some of the CIs might set up the environment using `requirements.txt` so I'd rather keep it.

**Comment by dominikwelke** [comment]
> A bit confused about that one. You've already committed your suggestions, right?

yes, i was confused too :D 
my *intention* was to suggest changes, but then i *did* a normal commit.. 
can you remind me how to suggest changes via the github website? ppl in the mne prs do it all the time, but i didnt find it earlier.

> `conda` won't automatically install stuff from `requirements.txt` so there won't be any double installation. I am not entirely sure but I think some of the CIs might set up the environment using `requirements.txt` so I'd rather keep it.

alright, my bad.. i was reading to fast - i thought it would pip install the `requirements.txt` in root as well 
```
  - pip:
      - -r file:freqtag/tests/requirements.txt
      - -r file:doc/requirements.txt
      - twine
```

**Comment by kalenkovich** [comment]
> yes, i was confused too :D
> my intention was to suggest changes, but then i did a normal commit..

Ah. alright. My intentions and my results often diverge as well when it comes to GitHub :) My first ever PR: just merged it myself immediately after creating it. We then decided to change some settings so I wouldn't be able to merge at all :)

> can you remind me how to suggest changes via the github website? ppl in the mne prs do it all the time, but i didnt find it earlier.

Sure! You go under the "Files changed" tab. There, you select a line or a couple of lines and press "+" to add a comment. In the comment toolbar above the editing area, there is a button with a plus-minus sign which allows you to add a suggestion alongside the comment.

**Comment by kalenkovich** [comment]
Since otherwise everything seems to work, I'll just fix the PEP8 issues and merge.

**Pull request merged**